clients/go/gengo.go

```diff
@@ -73,54 +90,246 @@ type WagClient struct {
+		// the ID generator must generate trace IDs uniformly at random from the entire space of uint64.
+		// For example, the default x-ray ID generator does not do this.
+		sdktrace.WithSampler(sdktrace.AlwaysSample()),
+		// These maximums are to guard against something going wrong and sending a ton of data unexpectedly
+		sdktrace.WithSpanLimits(sdktrace.SpanLimits{
+			AttributeCountLimit: 100,
+			EventCountLimit:     100,
+			LinkCountLimit:      100,
+		}),
+		//Batcher is more efficient, switch to it after testing
+		// sdktrace.WithSyncer(exporter),
+		sdktrace.WithBatcher(exporter),
+		sdktrace.WithResource(newResource()),
+		
+	)
+	otel.SetTracerProvider(tp)
+	otel.SetTextMapPropagator(propagation.NewCompositeTextMapPropagator(propagation.TraceContext{}, propagation.Baggage{}))
+	return tp
+}
+
+func doNothing(baseTransport http.RoundTripper, spanNameCtxValue interface{}, tp sdktrace.TracerProvider) http.RoundTripper {
+	return baseTransport
+}
+
+func determineSampling() (samplingProbability float64, err error) {
+
+		// If we're running locally, then turn off sampling. Otherwise sample
+		// 1%% or whatever TRACING_SAMPLING_PROBABILITY specifies.
+		samplingProbability = 0.01
+		isLocal := os.Getenv("_IS_LOCAL") == "true"
+		if isLocal {
+			fmt.Println("Set to Local")
+			samplingProbability = 1.0
+		} else if v := os.Getenv("TRACING_SAMPLING_PROBABILITY"); v != "" {
+			samplingProbabilityFromEnv, err := strconv.ParseFloat(v, 64)
+			if err != nil {
+				return 0, fmt.Errorf("could not parse '%%s' to float", v)
+			}
+			samplingProbability = samplingProbabilityFromEnv
+		}
+		return
+	}
+
+//----------------------END TRACING RELATEDFUNCTIONS----------------------
+
 // New creates a new client. The base path and http transport are configurable.
-func New(basePath string) *WagClient {
+func New(ctx context.Context, basePath string, opts ...Option) *WagClient {
+
+	defaultTransport := http.DefaultTransport
+	defaultLogger := NewLogger("{{.ServiceName}}-wagclient", wcl.Info)
+	defaultExporter := tracetest.NewNoopExporter()
+	defaultInstrumentor := doNothing
+
+
 	basePath = strings.TrimSuffix(basePath, "/")
 	base := baseDoer{}
 	// For the short-term don't use the default retry policy since its 5 retries can 5X
 	// the traffic. Once we've enabled circuit breakers by default we can turn it on.
 	retry := retryDoer{d: base, retryPolicy: SingleRetryPolicy{}}
-	logger := logger.New("{{.ServiceName}}-wagclient")
+	options := options{
+		transport:    defaultTransport,
+		logger:       defaultLogger,
+		exporter:     defaultExporter,
+		instrumentor: defaultInstrumentor,
+	}
+
+	for _, o := range opts {
+		o.apply(&options)
+	}
+
+
+	samplingProbability := 1.0 // TODO: Hard setting this to one for now, because right now 
+	// it is essentially ignored as the sidecar is determining the sample rate it forwards on to DD.
+	// samplingProbability := determineSampling()
```

Comment:
Remove the TODO and just explain that the preferred approach is to sample locally in the sidecar?